services/core-web/common/actionCreators/permitActionCreator.ts

```diff
@@ -0,0 +1,501 @@
+import { notification } from "antd";
+import { showLoading, hideLoading } from "react-redux-loading-bar";
+import {
+  ENVIRONMENT,
+  ICreatePermitPayload,
+  IPermit,
+  IPermitAmendment,
+  ICreatePermitAmendmentPayload,
+  ICreatePermitConditionPayload,
+  IPermitCondition,
+  IUpdatePermitAmendmentPayload,
+} from "@mds/common";
+import { request, success, error } from "../actions/genericActions";
+import * as reducerTypes from "../constants/reducerTypes";
+import * as permitActions from "../actions/permitActions";
+import * as String from "../constants/strings";
+import * as API from "../constants/API";
+import { createRequestHeader } from "../utils/RequestHeaders";
+import CustomAxios from "../customAxios";
+import { AxiosResponse } from "axios";
+import { AppThunk } from "@/store/appThunk.type";
+
+export const createPermit =
+  (mineGuid: string, payload: ICreatePermitPayload): AppThunk<Promise<AxiosResponse<IPermit>>> =>
+  (dispatch): Promise<AxiosResponse<IPermit>> => {
+    dispatch(request(reducerTypes.CREATE_PERMIT));
+    dispatch(showLoading("modal"));
+    return CustomAxios()
+      .post(ENVIRONMENT.apiUrl + API.PERMITS(mineGuid), payload, createRequestHeader())
+      .then((response: AxiosResponse<IPermit>) => {
+        notification.success({
+          message: "Successfully created a new permit",
+          duration: 10,
+        });
+        dispatch(success(reducerTypes.CREATE_PERMIT));
+        return response;
+      })
+      .catch((err) => {
+        dispatch(error(reducerTypes.CREATE_PERMIT));
+        throw new Error(err);
+      })
+      .finally(() => dispatch(hideLoading("modal")));
+  };
+
+export const fetchPermits = (mineGuid: string) => (dispatch) => {
```

Comment:
If an action creator isn't returning anything, you can still use the `AppThunk` type, but without the return type passed in.  This will make the dispatch aware of the global state object.
```suggestion
export const fetchPermits = (mineGuid: string): AppThunk => (dispatch) => {
```

Reply:
This has been adjusted now, thanks!